lib/board.dart

```diff
@@ -195,21 +194,45 @@ class _BoardState extends State<Board> with WidgetsBindingObserver {
   }
 
   void toggleMusic() {
+    if (!musicStarted) {
+      _assetsAudioPlayer.open(
+          Playlist(
+            audios: [
+              Audio("assets/audios/background_music.mp3"),
+            ],
+          ),
+          loopMode: LoopMode.playlist,
+          autoStart: true);
+      setState(() {
+        musicStarted = true;
+        isPlaying = true;
+      });
+      return;
+    }
     if (isPlaying) {
-      _assetsAudioPlayer.stop();
-      isPlaying = false;
+      _assetsAudioPlayer.playOrPause();
+      setState(() {
+        isPlaying = false;
+      });
     } else {
-      _assetsAudioPlayer.open(
-        Playlist(
-          audios: [
-            Audio("assets/audios/background_music.mp3"),
-          ],
-        ),
-        loopMode: LoopMode.playlist,
-        autoStart: true
-      );
+      _assetsAudioPlayer.playOrPause();
+      setState(() {
+        isPlaying = true;
+      });
+    }
+  }
 
-      isPlaying = true;
+  void toggleSound() {
```

Comment:
Please explain why we have this function?
If we have just the one button for music on and off